File: app-java/src/main/java/com/tritiumstudios/phasmophobiaevidencetool/android/firebase/firestore/transactions/user/account/FirestoreTransactionHistory.java
package com.tritiumstudios.phasmophobiaevidencetool.android.firebase.firestore.transactions.user.account;

import android.util.Log;

import androidx.annotation.NonNull;

import com.tritiumstudios.phasmophobiaevidencetool.android.firebase.firestore.transactions.user.FirestoreAccount;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.SetOptions;

import java.util.HashMap;
import java.util.Map;

public class FirestoreTransactionHistory {

    private final static String DOCUMENT_TRANSACTION_HISTORY = "TransactionHistory";

    public static void init() throws Exception {

        Map<Object, Object> emptyMap = new HashMap<>();

        getTransactionHistoryDocument()
            .get()
            .addOnSuccessListener(documentSnapshot -> {
                documentSnapshot.getReference().set(emptyMap, SetOptions.merge())
                        .addOnSuccessListener(unused ->
                                Log.d("Firestore",
                                        DOCUMENT_TRANSACTION_HISTORY +
                                                " successfully INITIALIZED!")
                        )
                        .addOnFailureListener(e -> {
                            Log.d("Firestore",
                                    DOCUMENT_TRANSACTION_HISTORY +
                                            " failed INITIALIZATION");
                            e.printStackTrace();
                        })
                        .addOnCompleteListener(task ->
                                Log.d("Firestore",
                                        DOCUMENT_TRANSACTION_HISTORY +
                                                " INITIALIZATION process complete!")
                        );
            })
            .addOnFailureListener(Throwable::printStackTrace);

    }

    @NonNull
    public static DocumentReference getTransactionHistoryDocument()
            throws Exception {

        return FirestoreAccount.getAccountCollection()
                .document(DOCUMENT_TRANSACTION_HISTORY);
    }

}

File: app-compose/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/core/data/market/palette/source/remote/api/firestore/transactions/user/account/transactions/FirestoreTransactionHistory.kt
package com.tritiumgaming.phasmophobiaevidencepicker.core.data.market.palette.source.remote.api.firestore.transactions.user.account.transactions

import android.util.Log
import com.google.android.gms.tasks.Task
import com.google.firebase.firestore.DocumentReference
import com.google.firebase.firestore.DocumentSnapshot
import com.google.firebase.firestore.SetOptions
import com.tritiumgaming.phasmophobiaevidencepicker.core.data.market.palette.source.remote.api.firestore.transactions.user.account.FirestoreAccount.Companion.accountCollection

class FirestoreTransactionHistory {

    companion object {
        private const val DOCUMENT_TRANSACTION_HISTORY = "TransactionHistory"

        @get:Throws(Exception::class)
        val transactionHistoryDocument: DocumentReference
            get() = accountCollection.document(DOCUMENT_TRANSACTION_HISTORY)

        @Throws(Exception::class)
        fun init() {
            val emptyMap: Map<Any, Any> = HashMap()

            transactionHistoryDocument
                .get()
                .addOnSuccessListener { documentSnapshot: DocumentSnapshot ->
                    documentSnapshot.reference.set(emptyMap, SetOptions.merge())
                        .addOnSuccessListener { unused: Void? ->
                            Log.d(
                                "Firestore",
                                DOCUMENT_TRANSACTION_HISTORY +
                                        " successfully INITIALIZED!"
                            )
                        }
                        .addOnFailureListener { e: Exception ->
                            Log.d(
                                "Firestore",
                                DOCUMENT_TRANSACTION_HISTORY +
                                        " failed INITIALIZATION"
                            )
                            e.printStackTrace()
                        }
                        .addOnCompleteListener { task: Task<Void?>? ->
                            Log.d(
                                "Firestore",
                                DOCUMENT_TRANSACTION_HISTORY +
                                        " INITIALIZATION process complete!"
                            )
                        }
                }
                .addOnFailureListener { obj: Exception -> obj.printStackTrace() }
        }

    }
}
